### PR TITLE
Issue #3: Unit tests added to server

### PR DESCRIPTION
**Pull request by Mahesh419**
This PR contains following work.
- Setup mocha test runner with chai and mockgoose for unit testing in server
- Added nyc for code coverage report with a custom script 'coverage'
- Added unit test cases for all leaderboard GET endpoints

### CODE REVIEW

File: server/.env [hunk 1]
```diff
@@ -1,2 +1,2 @@
-MONGO_STR=<database_string>
+MONGO_STR=mongodb://admin.localhost:27017/walk2win-test
```
Comment (shalithasuranga) [inline server/.env]: Can you please keep mongo connection string as `<database_string>` because dev will use what they want :)

Comment (Mahesh419) [reply server/.env]: sure I will update it. It was a mistake.

Comment (Mahesh419) [reply server/.env]: Updated. :+1:

**Comment by shalithasuranga** [comment]
@Mahesh419  could you please resolve the conflict . So we can merge this now

**Pull request merged**